### PR TITLE
Add `packages.${system}.haddock` attribute for combined haddock

### PR DESCRIPTION
**Pull request by L-as**
Unfortunately, it seems that we can't add all packages to this,
it errs with `Couldn't get the project and version from the html`,
and I'm not sure why.

Do `nix build .#haddock` to test. This is self-contained and can be
hosted on GitHub Pages.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -479,16 +479,61 @@
...
+                * "Plutarch.Prelude"
+                * "Plutarch"
+            '';
+          };
+        };
     in
     {
       inherit extraSources cabalProjectLocal haskellModule tools;
 
+      nixpkgsDebug = perSystem nixpkgsFor;
```
Comment (srid) [inline flake.nix:527]: ?

Comment (L-as) [reply flake.nix:527]: Thank you for catching this!

**Comment by L-as** [comment]
When this is merged, the documentation should be available on https://plutonomicon.github.io/plutarch/

**Comment by TotallyNotChase** [comment]
Is this just for haddock or is it possible to get hoogle going as well?

**Comment by L-as** [comment]
It's just for Haddock. Hoogle would require a ton of effort to get running I think, and I'm not even sure it works without a server.

**Comment by srid** [comment]
Running Hoogle is pretty straightforward in Nix, but we need a place to run it, and deploy it via CI/CD.

CD can be obviated with https://docs.cachix.org/deploy/ - and should work on local machines as well. Cachix Deploy however was in beta, last I checked. So YMMV.

**Pull request merged**